src/main/java/pl/antonina/tasks/child/ChildService.java

```diff
@@ -34,11 +38,16 @@ ChildView getChild(long id) {
     }
 
     void addChild(long parentId, ChildData childData) {
-        Parent parent = parentRepository.findById(parentId).orElseThrow();
+        String email = childData.getUserData().getEmail();
+        boolean userExists = userRepository.findByEmail(email).isPresent();
+        if (userExists) {
+            throw new IllegalArgumentException("User with given email already exists");
```

Comment:
Please consider:

using own type of exception that will be mapped later to proper exception or directly the error code (https://www.baeldung.com/exception-handling-for-rest-with-spring)
adding throws IllegalArgumentException (or your own Exception name) to the method signature - it's part of this method flow, something that is quite expected. :)

Reply:
I can agree :)

Reply:
Done :)